machine-learning/ml_tips/docs/more.md

```diff
@@ -0,0 +1,718 @@
+      , "${reg}" as reg
+    from
+      prediction p
+    ;
+
+This query inserts RMSE (Root Mean Squared Error) and MAE (Mean Absolute Error), measurements for regression, to find the best parameters.
+
+Now, you can find the best model by using the following workflow.
+
+    +persist_best:
+      +choose_best:
+        td>: queries/best_param.sql
+        engine: presto
+        measure: rmse
+        store_last_results: true
+    
+      +save_model:
+        td_ddl>:
+        suffix: _${td.last_results.reg}_${td.last_results.eta0.toString().replace('.', '_')}
+        rename_tables: [{from: "regressor${suffix}", to: regressor_best}]
+
+    -- Presto: queries/best_param.sql
+    select
+      *
+    from
+      accuracy_test
+    order by
+      ${measure}
+    limit 1
+    ;
+
+Finally, `+save_model` task saves the best model as `regressor_best` table.
+
+Here is the whole workflow for a grid search. This workflow and the queries can be found at the [workflow-examples repository](https://github.com/treasure-data/workflow-examples/blob/master/machine-learning/house_price/search_hyperparams.dig).
+
+    +parameter_tuning:
+      for_each>:
+        eta0: [5.0, 1.0, 0.5, 0.1, 0.05, 0.01, 0.001]
+        reg: ['no', 'rda', 'l1', 'l2', 'elasticnet']
+    
+      _parallel: true
+    
+      _do:
+        +train:
+          td>: queries/train_regressor.sql
+          suffix: _${reg}_${eta0.toString().replace('.', '_')}
+          create_table: regressor${suffix}
+    
+        +evaluate:
+          td>: queries/evaluate_params.sql
+          insert_into: accuracy_test
+          suffix: _${reg}_${eta0.toString().replace('.', '_')}
+    
+    +parsist_best:
+      +choose_best:
+        td>: queries/best_param.sql
+        engine: presto
+        measure: rmse
+        store_last_results: true
+    
+      +show_accuracy:
+        echo>: "Best model reg: ${td.last_results.reg}, eta0: ${td.last_results.eta0} RMSE: ${td.last_results.rmse}, MAE: ${td.last_results.mae}"
+    
+      +save_model:
+        td_ddl>:
+        suffix: _${td.last_results.reg}_${td.last_results.eta0.toString().replace('.', '_')}
+        rename_tables: [{from: "regressor${suffix}", to: regressor_best}]
+    
+      +clean_up:
+        for_each>:
+          eta0: [5.0, 1.0, 0.5, 0.1, 0.05, 0.01, 0.001]
+          reg: ['no', 'rda', 'l1', 'l2', 'elasticnet']
+    
+        _parallel: true
+    
+        _do:
+          +drop_table:
+            td_ddl>:
+            suffix: _${reg}_${eta0.toString().replace('.', '_')}
+            drop_tables: ["regressor${suffix}"]
+
+### Random search
+
+*This approach requires Python custom scripting, currently under beta feature.*
+
+[Random search](https://en.wikipedia.org/wiki/Random_search) is also known as [an option for hyperparameter search](http://www.jmlr.org/papers/v13/bergstra12a.html). You can run a random search with `py>` operator as follows:
+
+    # random_search.dig
+    +generate_param_list:
+      py>: rand.rand_params
```

Comment:
assuming `py>` operator. 

Better to add note `py>` is not supported in TD yet.

Reply:
I think this note is enough, but should we add more warning?
https://github.com/treasure-data/workflow-examples/pull/160/files#diff-5fc6f49c257bd73144b71efe39158500R651

Reply:
Corrected with more appropriate description 435e2c83a5a641b12c805d0f865ba5416af09ddb